openmc/model/triso.py

```diff
@@ -153,3 +538,490 @@ def create_triso_lattice(trisos, lower_left, pitch, shape, background):
+                         '"cylinder", and "sphere" are '
+                         'supported."'.format(domain_shape))
+    if not domain_length and domain_shape in ['cube', 'cylinder']:
+        raise ValueError('"domain_length" must be specified for {} domain '
+                         'geometry '.format(domain_shape))
+    if not domain_radius and domain_shape in ['cylinder', 'sphere']:
+        raise ValueError('"domain_radius" must be specified for {} domain '
+                         'geometry '.format(domain_shape))
+
+    if domain_shape is 'cube':
+        domain = _CubicDomain(length=domain_length, particle_radius=radius,
+                              center=domain_center)
+    elif domain_shape is 'cylinder':
+        domain = _CylindricalDomain(length=domain_length, radius=domain_radius,
+                                    particle_radius=radius, center=domain_center)
+    elif domain_shape is 'sphere':
+        domain = _SphericalDomain(radius=domain_radius, particle_radius=radius,
+                                  center=domain_center)
+
+    # Calculate the packing fraction if the number of particles is specified;
+    # otherwise, calculate the number of particles from the packing fraction.
+    if ((n_particles is None and packing_fraction is None) or
+        (n_particles is not None and packing_fraction is not None)):
+        raise ValueError('Exactly one of "n_particles" and "packing_fraction" '
+                         'must be specified.')
+    elif packing_fraction is None:
+        n_particles = int(n_particles)
+        packing_fraction = 4/3*pi*radius**3*n_particles / domain.volume
+    elif n_particles is None:
+        packing_fraction = float(packing_fraction)
+        n_particles = int(packing_fraction*domain.volume // (4/3*pi*radius**3))
+
+    # Check for valid packing fractions for each algorithm
+    if packing_fraction >= 0.64:
+        raise ValueError('Packing fraction of {} is greater than the '
+                         'packing fraction limit for close random '
+                         'packing (0.64)'.format(packing_fraction))
+    if initial_packing_fraction >= 0.38:
+        raise ValueError('Initial packing fraction of {} is greater than the '
+                         'packing fraction limit for random sequential'
+                         'packing (0.38)'.format(initial_packing_fraction))
+    if initial_packing_fraction > packing_fraction:
+        initial_packing_fraction = packing_fraction
+        if packing_fraction > 0.3:
+            initial_packing_fraction = 0.3
+
+    random.seed(seed)
+
+    # Set parameters for initial random sequential packing of particles.
+    initial_radius = (3/4 * initial_packing_fraction * domain.volume /
+                      (pi * n_particles))**(1/3)
+    domain.particle_radius = initial_radius
+    domain.limits = [[x - initial_radius + radius for x in domain.limits[0]],
+                     [x + initial_radius - radius for x in domain.limits[1]]]
```

Comment:
If random sequential packing is used to initialize a particle configuration for close random packing, the radius of the particles in the RSP is smaller than the desired final particle radius. The limits for RSP are recalculated here using the final particle radius instead of the initial radius so the particles are fully contained within the domain when the CRP begins. I should have clarified that — will add it in the comments.